src/lib/y2partitioner/widgets/format_and_mount.rb

```diff
@@ -249,12 +248,62 @@ def handle(event)
         nil
       end
 
+      # @see #errors
       def validate
-        validate_system_mount_points && validate_mount_by_label
+        return false unless system_mount_points_warning
+
+        error = errors.first
+
+        if error
+          Yast2::Popup.show(error, headline: :error)
+          return false
+        end
+
+        btrfs_subvolumes_question
+        true
       end
 
       private
 
+      # @return [Actions::Controllers::Filesystem]
+      attr_reader :controller
+
+      # Asks to the user what to do with the current list of Btrfs subvolumes
+      #
+      # Note that the user is only asked when the Btrfs filesystem does not exist on disk yet and it has
+      # subvolumes.
+      def btrfs_subvolumes_question
+        return unless controller.new_btrfs? && controller.mount_path_modified?
+
+        controller.restore_btrfs_subvolumes = true
+
+        # It does not have subvolumes, so let's simply restore the default list.
+        return unless controller.btrfs_subvolumes?
+
+        if controller.default_btrfs_subvolumes?
+          # TRANSLATORS: message to ask to the user about restoring the default list of subvolumes.
+          message = _("Do you want to restore the default list of subvolumes?")
```

Comment:
I find this message to be too succinct. Shouldn't we explain the situation a bit better?

Something in the line of:

```
You have chosen to mount the device at #{mount_point}.

Do you want to remove the current subvolumes and
create the suggested subvolumes for #{mount_point}?
```